src/Check/DoctrineMigration.php

```diff
@@ -17,70 +17,94 @@
+                If you are using doctrine/migrations ^3.0,
+                pass Doctrine\Migrations\DependencyFactory as argument.
+                If you are using doctrine/migrations ^2.0,
+                pass Doctrine\Migrations\Configuration\Configuration as argument.
+                MESSAGE
+            );
         }
 
-        // check for doctrine/migrations:^2.0
-        if ($input instanceof Configuration
-            && method_exists($input, 'getAvailableVersions')
-            && method_exists($input, 'getMigratedVersions')
-        ) {
-            $this->availableVersions = $input->getAvailableVersions();
-            $this->migratedVersions = $input->getMigratedVersions();
-            return;
-        }
-
-        throw new InvalidArgumentException(<<<'MESSAGE'
-            Invalid Argument for DoctrineMigration check.
-            If you are using doctrine/migrations ^3.0, pass Doctrine\Migrations\DependencyFactory as argument.
-            If you are using doctrine/migrations ^2.0, pass Doctrine\Migrations\Configuration\Configuration as argument.
-            MESSAGE
-        );
+        $this->input = $input;
     }
 
     /**
      * Perform the actual check and return a ResultInterface
      */
     public function check(): ResultInterface
     {
-        $notMigratedVersions = array_diff($this->availableVersions, $this->migratedVersions);
+        $availableVersions = $this->getAvailableVersions();
+        $migratedVersions = $this->getMigratedVersions();
+
+        $notMigratedVersions = array_diff($availableVersions, $migratedVersions);
         if (! empty($notMigratedVersions)) {
             return new Failure('Not all migrations applied', $notMigratedVersions);
         }
 
-        $notAvailableVersion = array_diff($this->migratedVersions, $this->availableVersions);
+        $notAvailableVersion = array_diff($migratedVersions, $availableVersions);
         if (! empty($notAvailableVersion)) {
             return new Failure('Migrations applied which are not available', $notMigratedVersions);
         }
 
         return new Success();
     }
 
+    /**
+     * @return Version[]
+     */
+    private function getAvailableVersions(): array
+    {
+        if ($this->input instanceof DependencyFactory) {
+            return $this->getAvailableVersionsFromDependencyFactory($this->input);
+        }
+
+        if ($this->input instanceof Configuration) {
+            return $this->input->getAvailableVersions();
+        }
+
+        throw new LogicException(<<<'MESSAGE'
+            Unexpected class for $input detected. Could not check for Doctrine Migrations.
+            MESSAGE
+        );
+    }
+
+    /**
+     * @return Version[]
+     */
+    private function getMigratedVersions(): array
+    {
+        if ($this->input instanceof DependencyFactory) {
+            return $this->getMigratedVersionsFromDependencyFactory($this->input);
+        }
+
+        if ($this->input instanceof Configuration) {
```

Comment:
This conditional can be removed, see https://github.com/laminas/laminas-diagnostics/pull/20/files#r569752752